### PR TITLE
updating to feb 2023 preview

### PR DESCRIPTION
**Pull request by karishmadaga**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json [hunk 1]
```diff
@@ -12077,7 +14482,7 @@
           },
           {
             "value": "RandomForest",
-            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the “bagging” method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
+            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the 'bagging' method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
```
Comment (jianyexi) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:14485]:
```suggestion
            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the 'bagging' method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
```

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json [hunk 2]
```diff
@@ -9849,7 +11652,7 @@
...
```
Comment (jianyexi) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:11655]:
```suggestion
            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the 'bagging' method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
```

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json [hunk 3]
```diff
@@ -16355,7 +19354,7 @@
...
```
Comment (jianyexi) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:19357]:
```suggestion
            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the 'bagging' method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
```
```suggestion
            "description": "Random forest is a supervised learning algorithm.\nThe \"forest\" it builds, is an ensemble of decision trees, usually trained with the 'bagging' method.\nThe general idea of the bagging method is that a combination of learning models increases the overall result."
```

**Comment by zizw123** [comment]
Please get breaking change reviewed and then come back for ARM review.

**Comment by karishmadaga** [comment]
> Please get breaking change reviewed and then come back for ARM review.

The change was approved in the previous API version by the ARM Breaking Changes board here: https://github.com/Azure/azure-rest-api-specs/pull/21761#issuecomment-1358739945

**Comment by karishmadaga** [comment]
For Swagger ModelValidation Errors for the missing required property 'secrets' -- See previous discussion:
https://github.com/Azure/azure-rest-api-specs/issues/20235
FYI @ArcturusZhang

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json [hunk 4]
```diff
@@ -10461,6 +12958,18 @@
       },
       "additionalProperties": false
     },
+    "ComputeResource": {
```
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:12961]: You add this model here, but other swagger files already contain model with the same name, and it has completely different properties.
This breaks the SDK generation and might produce ill-named SDKs. Please change the name of this model to something else.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json [hunk 5]
```diff
@@ -9384,6 +11796,41 @@
       "x-ms-client-name": "BatchDeploymentProperties",
       "additionalProperties": false
     },
+    "BatchDeploymentProperties": {
```
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:11799]: Same as `ComputeResource`, other swagger file already contains the model with the same name but with completely different properties.
PLease change a name for this model.

Comment (karishmadaga) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:11799]: I only see this x-ms-client name using the "BatchDeploymentProperties" name for the previous version: https://github.com/Azure/azure-rest-api-specs/blob/653f6cdde3728e5021229d1d78d8699e41d70600/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-12-01-preview/mfe.json#L9384

**Comment by rkmanda** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/registries/{registryName}/data": {

can this not be called containers or dataContainers?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1206 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Comment by rkmanda** [comment]
>             "name": "listViewType",

please use odata filters instead

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1239 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Comment by rkmanda** [comment]
>             "name": "$orderBy",

use odata instead by specifing $filter query param

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1550 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Comment by rkmanda** [comment]
>             "name": "$orderBy",

you can find existing examples in the rest-api-specs-repo

---
In reply to: [1434199275](https://github.com/Azure/azure-rest-api-specs/pull/22456#issuecomment-1434199275) [](http://example.com/codeflow?ancestors=1434199275)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1550 in 663a7a5. [](commit_id = 663a7a555b9b8c162a068519256f3a6563c9e147, deletion_comment = False)

**Comment by rkmanda** [comment]
>           "x-nullable": true

what does it mean for this property to have both x-ms-mutability create and nullable:true? If this property is not specified during create, it can never ever be specified again. Is that your intention?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:19822 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Comment by wmengmsft** [comment]
>

**Comment by rkmanda** [comment]
>             "name": "listViewType",

resolving this comment as the concepts are being kept similar to other resource types in the same namespace.

---
In reply to: [1434192868](https://github.com/Azure/azure-rest-api-specs/pull/22456#issuecomment-1434192868) [](http://example.com/codeflow?ancestors=1434192868,1434192868)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1239 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Comment by rkmanda** [comment]
>             "name": "$orderBy",

resolving this comment as the concepts are being kept similar to other resource types in the same namespace.

---
In reply to: [1434200227](https://github.com/Azure/azure-rest-api-specs/pull/22456#issuecomment-1434200227) [](http://example.com/codeflow?ancestors=1434200227,1434199275)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-02-01-preview/mfe.json:1550 in bc9aad6. [](commit_id = bc9aad601b2185b940b7cb221e825395b42d68b5, deletion_comment = False)

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**